core/transaction/validator.go

```diff
@@ -348,6 +379,33 @@ func CheckTransactionPayload(txn *Transaction) error {
 		if name == nil {
 			return errors.New(fmt.Sprintf("no name registered for pubKey %+v", pld.Registrant))
 		}
+	case *payload.Subscribe:
+		duration := pld.Duration
+		if duration > MaxSubscriptionDuration {
+			return errors.New(fmt.Sprintf("subscription duration %d can't be bigger than %d", duration, MaxSubscriptionDuration))
+		}
+
+		topic := pld.Topic
+		match, err := regexp.MatchString("([a-z]{8,12})", topic)
```

Comment:
should we give a little bit more flexibility in topic name? like a hyphen and a wider length limit?

Reply:
no problem, just say the values

Reply:
This is Google Cloud Pub/Sub's restrictions for reference. It's url friendly and thus quite safe.

- Start with a letter
- Contain between 3 and 255 characters
- Contain only the following characters:
  - Letters: [A-Za-z]
  - Numbers: [0-9]
  - Dashes: -
  - Underscores: _
  - Periods: .
  - Tildes: ~
  - Plus signs: +
  - Percent signs: %

Reply:
done